paddle/fluid/framework/lod_tensor.cc

```diff
@@ -418,7 +418,7 @@ void LoDTensor::MergeLoDTensor(
     PADDLE_ENFORCE_EQ(new_lod.size(), lod.size());
     for (size_t j = 0; j < lod.size(); ++j) {
       auto &sub_lod = new_lod[j];
-      auto &offset = sub_lod.back();
+      size_t offset = sub_lod.back();
```

Comment:
Don't use `reference` here, because `sub_lod` is `std::vector` and  `sub_lod.push_back` may change the address of the element of `sub_lod`.